engine/schema/src/com/cloud/upgrade/DatabaseUpgradeChecker.java

```diff
@@ -424,27 +421,9 @@ protected void upgrade(CloudStackVersion dbVersion, CloudStackVersion currentVer
                 }
 
                 upgrade.performDataMigration(conn);
-                boolean upgradeVersion = true;
-
-                if (upgrade.getUpgradedVersion().equals("2.1.8")) {
```

Comment:
I'm not sure about the implications, but since 2.x is a very old version -- we need to put in our release notes that we won't support upgrades from very old 2.x releases. We can in that case even remove all 2.x upgrade paths.